future-apps/payroll/contracts/Payroll.sol

```diff
@@ -594,21 +595,54 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
         emit TerminateEmployee(_employeeId, employee.accountAddress, _endDate);
     }
 
-    function _getOwedSalary(uint256 _employeeId, uint64 _date) internal view returns (uint256) {
+    /**
+     * @dev Calculates the date timestamp corresponding to the requested paying amount based on the employee's last payroll date.
+     * @param _employeeId Employee's identifier.
+     * @param _payedAmount Amount payed to the employee to query the timestamp of.
+     * @return The date timestamp in seconds corresponding to the given paying amount based on the employee's last payroll date.
+     */
+    function _getLastPayroll(uint256 _employeeId, uint256 _payedAmount) internal view returns (uint64) {
+        Employee storage employee = employees[_employeeId];
+
+        uint256 timeDiff = _payedAmount.div(employee.denominationTokenSalary);
+
+        // This function is only called from _payday, where we make sure that _payedAmount is lower or equal to the
+        // total owed amount, that is obtained from _getOwedSalary, which does exactly the opposite calculation:
+        // multiplying the employee's salary by an uint64 number of seconds. Therefore, timeDiff will always fit in 64.
+        // Nevertheless, we are performing a sanity check at the end to ensure the computed last payroll timestamp
+        // is not greater than the current timestamp.
+
+        uint256 lastPayrollDate = employee.lastPayroll.add(timeDiff);
+        require(lastPayrollDate <= uint256(getTimestamp64()), ERROR_LAST_PAYROLL_DATE_TOO_BIG);
```

Comment:
I still think here we can just use `getTimestamp()`.

Reply:
We are, see [here](https://github.com/aragon/aragon-apps/pull/744/files#diff-9d7adaffbcce19867f027a983a30bcc7R610). I think you were reviewing an old version :)